src/libprojectM/Renderer/PerlinNoiseWithAlpha.hpp

```diff
@@ -0,0 +1,176 @@
+/*
+ * PerlinNoise.hpp
+ *
+ *  Created and based on PerlinNoise.hpp
+ *  Created on: Sep 14, 2019
+ *      Author: hibengler
+ */
+
+
+/* The reason for this cousin class is that in Open GLES 2.0 environments,
+the glTexImage2d cannot convert from GL_RGB into GL_RGBA
+so the TextureManager has to use something that is pre-RGBA
+*/
+#ifndef PERLINNOISEWITHALPHA_HPP_
+#define PERLINNOISEWITHALPHA_HPP_
+
+#include <math.h>
+
+class PerlinNoiseWithAlpha
+{
+public:
+
+    float noise_lq[256][256][4];
+    float noise_lq_lite[32][32][4];
+    float noise_mq[256][256][4];
+    float noise_hq[256][256][4];
+    float noise_lq_vol[32][32][32][4];
+    float noise_hq_vol[32][32][32][4];
+
+
+	PerlinNoiseWithAlpha();
+	virtual ~PerlinNoiseWithAlpha();
+
+private:
+
+	static inline float noise( int x)
+	{
+	    x = (x<<13)^x;
```

Comment:
i truly actually have no idea what is going on here. comment? make some constants?

Reply:
No, it is generating random noise data used by some of the visualizers. 

The noise functions convert the standard random numbers to follow different 
ranges, styles, etc.

inline float noise looks to be generating a psuedo random number that is seeded with X.


 This file is basically a copy of PerlInNoise.hpp, and does the same thing as PerlInNoise class, except that it generates the noise in an RGBA buffer instead of an RGB buffer.

OpenGL ES 2.0 cannot transform RGB into RGBA in hardware,   but it can do RGBA to RGBA.  This is why I me the step-sibling class.